### PR TITLE
Custom Event Hub Endpoint

### PR DESCRIPTION
**Pull request by formulahendry**
* Monitor Custom Event Hub Endpoint
* Endpoints tree view
* Fix security vulnerability

New tree view:
image

Better naming:
image
image
image
image
image
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -260,6 +267,16 @@
         "command": "azure-iot-toolkit.copyModuleConnectionString",
         "title": "Copy Module Connection String",
         "category": "Azure IoT Hub"
+      },
+      {
+        "command": "azure-iot-toolkit.startMonitorCustomEventHubEndpoint",
+        "title": "Start Monitoring",
```
Comment (adashen) [inline package.json]: title same as line 94?

File: src/Nodes/Endpoints/EndpointsLabelNode.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+            const subscription = accountApi.subscriptions.find((element) => element.subscription.subscriptionId === subscriptionId);
+            const client = new IotHubClient(subscription.session.credentials, subscription.subscription.subscriptionId, subscription.session.environment.resourceManagerEndpointUrl);
+            const iotHubs = await client.iotHubResource.listBySubscription();
+            const iothub = iotHubs.find((element) =>
+                element.id === Constants.ExtensionContext.globalState.get(Constants.StateKeyIoTHubID));
+            TelemetryClient.sendEvent(Constants.IoTHubAILoadEndpointsTreeDoneEvent, { Result: "Success" });
+
+            return [new BuiltInEndpointLabelNode(),
+                new EventHubLabelNode(subscription, iothub.properties.routing.endpoints.eventHubs),
```
Comment (adashen) [inline src/Nodes/Endpoints/EndpointsLabelNode.ts]: shall we check iothub before using it? It seems it is possible no iothub is got?

File: src/eventHubManager.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        }
+
+        try {
+            TelemetryClient.sendEvent(Constants.IoTHubAIEHStartMonitorEvent);
+            this._outputChannel.show();
+            this.outputLine(Constants.EventHubMonitorLabel, `Start monitoring message arrived in custom Event Hub endpoint [${eventHubItem.eventHubProperty.name}] ...`);
+
+            const eventHubClient = new EventHubManagementClient(eventHubItem.azureSubscription.session.credentials, eventHubItem.azureSubscription.subscription.subscriptionId);
+            const connectionString = (await eventHubClient.namespaces.listKeys(eventHubItem.eventHubProperty.resourceGroup,
+                this.getNamespacefromConnectionString(eventHubItem.eventHubProperty.connectionString), "RootManageSharedAccessKey")).primaryConnectionString;
```
Comment (adashen) [inline src/eventHubManager.ts]: So to monitor the message must have "RootManageSharedAccessKey" role?

File: src/eventHubManager.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+export class EventHubManager extends IoTHubMessageBaseExplorer {
+    private _eventHubClient: EventHubClient;
+
+    constructor(outputChannel: vscode.OutputChannel) {
+        super(outputChannel, "$(primitive-square) Stop Monitoring Custom Event Hub Endpoint", "azure-iot-toolkit.stopMonitorCustomEventHubEndpoint");
+    }
+
+    public async startMonitorCustomEventHubEndpoint(eventHubItem: EventHubItem) {
+        if (this._isMonitoring) {
```
Comment (adashen) [inline src/eventHubManager.ts]: build in endpoint and custom endpoint could be moniored at the same time?

Comment (formulahendry) [reply src/eventHubManager.ts]: As we synced offline, yes.

File: package.json (hunk 1 above)
Comment (formulahendry) [reply package.json]: Yes~

File: src/eventHubManager.ts (hunk 3 above)
Comment (formulahendry) [reply src/eventHubManager.ts]: Yes, it is similar to the concept of 'iothubowner' in IoT Hub. Though user could add his own, but we use this directly.

File: src/Nodes/Endpoints/EndpointsLabelNode.ts (hunk 2 above)
Comment (formulahendry) [reply src/Nodes/Endpoints/EndpointsLabelNode.ts]: Good catch!

**Pull request merged**